src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,49 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int PASSWORD_MIN_LENGTH = 6;
+    private static final int LOGIN_MIN_LENGTH = 6;
+    private static final int MIN_AGE = 18;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidUserDataException("Provided User is null");
+        }
+        validateLogin(user.getLogin());
+        validatePassword(user.getPassword());
+        validateAge(user.getAge());
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidUserDataException(
+                    "User with such login has been already registered. Login: "
+                    + user.getLogin());
+        }
+        return storageDao.add(user);
+    }
+
+    private void validateLogin(String login) {
+        if (login == null || login.length() < LOGIN_MIN_LENGTH) {
+            throw new InvalidUserDataException("Invalid User login was provided: "
+                    + login + ". Login can't be null and be less than "
+                    + LOGIN_MIN_LENGTH + " characters");
+        }
+    }
+
+    private void validatePassword(String password) {
+        if (password == null || password.length() < PASSWORD_MIN_LENGTH) {
+            throw new InvalidUserDataException("Invalid User password was provided: "
+                    + password + ". Password can't be null and be less than "
+                    + PASSWORD_MIN_LENGTH + " characters");
+        }
+    }
+
+    private void validateAge(Integer age) {
+        if (age == null || age < MIN_AGE) {
+            throw new InvalidUserDataException("Invalid User age was provided: "
+                    + age + ". Age can't be null or less then " + MIN_AGE);
+        }
```

Comment:
Don’t create one big if with all conditions
![image](https://github.com/mate-academy/jv-registration-validation/assets/135730971/1ba79e23-c80d-4a75-807d-bb2cfa6e9be7)

Reply:
In the 'bad example' we have overloaded 'if', which contains checks for all fields. In my solution, I have divided logic and a specific informative exception message for each field. Also, consider, that 'good example' doesn't cover all possible cases. Therefore we shouldn't just copy the example.

Reply:
I meant that you could display two messages about null and about the length, but you can leave it as it was.

Reply:
Got it and changed the conditions. Thanks!